### PR TITLE
Add divisibility in `ZLat`

### PR DESCRIPTION
**Pull request by StevellM**
I do not know whether it is the most efficient way, but it should do the job.

### CODE REVIEW

**Comment by thofma** [comment]
What is the divisibility of something? Can you share a reference?

**Comment by StevellM** [comment]
Hum I do not have a reference. If `v` is a primitive vector in a `ZLat` `(L,b)`, then the divisibility of `v` in `(L, b)` is the positive generator of the $\mathbb Z$-ideal $b(v, L)$.

**Comment by thofma** [comment]
Could you add this sentence to the documentation?

File: src/QuadForm/Quad/ZLattices.jl [hunk 1]
```diff
@@ -1392,13 +1393,63 @@ end
...
+
+For a primitive vector `v` in a $\mathbb Z$-lattice `L`, we
+call the divisibility of `v` in `L` the positive generator
+of the $\mathbb Z$-ideal $b(v, L)$, where `b` is the bilinear
+form associated to `L`.
+"""
+divisibility(::ZLat, ::Union{Vector, fmpq_mat})
+
+function divisibility(L::ZLat, v::Vector{<: RationalUnion})
+  @req is_primitive(L, v) "v must be a primitive vector in L"
```
Comment (simonbrandhorst) [inline src/QuadForm/Quad/ZLattices.jl:1440]: No need to assume primitive.
```suggestion
Return the divisibility of `v` with respect to `L`.

For a vector `v` in in the ambient space $(V,\Phi)$ of `L`, we
call the divisibility of `v` with respect to `L` the nonnegative generator
of the fractional $\mathbb Z$-ideal $\Phi(v, L)$.
"""
divisibility(::ZLat, ::Union{Vector, fmpq_mat})

function divisibility(L::ZLat, v::Vector{<: RationalUnion})
```

**Pull request merged**